src/self/subResourceWrapper/deploy.bicep

```diff
@@ -91,6 +91,159 @@ param roleAssignments array = []
 @sys.description('Disable telemetry collection by this module. For more information on the telemetry collected by this module, that is controlled by this parameter, see this page in the wiki: [Telemetry Tracking Using Customer Usage Attribution (PID)](https://github.com/Azure/bicep-lz-vending/wiki/Telemetry)')
 param disableTelemetry bool = false
 
+@maxLength(90)
+@sys.description('The name of the resource group to create the deployment script for resource providers registration.')
+param deploymentScriptResourceGroupName string
+
+@sys.description('The location of the deployment script. Use region shortnames e.g. uksouth, eastus, etc.')
+param deploymentScriptLocation string = deployment().location
+
+@sys.description('The name of the deployment script to register resource providers')
+param deploymentScriptName string
+
+@sys.description('''
+An object of resource providers and resource providers features to register. If left blank/empty, no resource providers or features will be registered.
```

Comment:
The "If left blank/empty" part of the description is not true, can we remove or reword?

We should register the list of providers by default